### ISSUE TITLE
Plugin crashes when building

### ISSUE THREAD

**Issue by FranseFrikandel** [captn3m0/jekyll-sqlite#1]
Hello there!

I was trying to use this plugin to read some data out of a sqlite database, however trying to build the site with the plugin enabled causes the following error:

```
Configuration file: C:/Users/CRGam/Documents/Metal Surface Treatment/Websites/static-sites/restometal/_config_dev.yml
            Source: C:/Users/CRGam/Documents/Metal Surface Treatment/Websites/static-sites/restometal
       Destination: //wsl$/Ubuntu/var/www/html
 Incremental build: disabled. Enable with --incremental
      Generating...
C:/Ruby31-x64/lib/ruby/gems/3.1.0/gems/jekyll-sqlite-0.1.2/lib/jekyll-sqlite/generator.rb:27:in `gen_hash_data': undefined local variable or method `name' for #<JekyllSQlite::Generator:0x000002243ef1ae28> (NameError)

      root[name] = db.execute(config["query"])
           ^^^^
        from C:/Ruby31-x64/lib/ruby/gems/3.1.0/gems/jekyll-sqlite-0.1.2/lib/jekyll-sqlite/generator.rb:62:in `gen_data'
        from C:/Ruby31-x64/lib/ruby/gems/3.1.0/gems/jekyll-sqlite-0.1.2/lib/jekyll-sqlite/generator.rb:80:in `block (2 levels) in generate'
        from C:/Ruby31-x64/lib/ruby/gems/3.1.0/gems/sqlite3-1.6.3-x64-mingw-ucrt/lib/sqlite3/database.rb:110:in `initialize'
        from C:/Ruby31-x64/lib/ruby/gems/3.1.0/gems/jekyll-sqlite-0.1.2/lib/jekyll-sqlite/generator.rb:76:in `new'
        from C:/Ruby31-x64/lib/ruby/gems/3.1.0/gems/jekyll-sqlite-0.1.2/lib/jekyll-sqlite/generator.rb:76:in `block in generate'
        from C:/Ruby31-x64/lib/ruby/gems/3.1.0/gems/jekyll-sqlite-0.1.2/lib/jekyll-sqlite/generator.rb:74:in `each'
        from C:/Ruby31-x64/lib/ruby/gems/3.1.0/gems/jekyll-sqlite-0.1.2/lib/jekyll-sqlite/generator.rb:74:in `generate'
        from C:/Ruby31-x64/lib/ruby/gems/3.1.0/gems/jekyll-4.3.2/lib/jekyll/site.rb:193:in `block in generate'
        from C:/Ruby31-x64/lib/ruby/gems/3.1.0/gems/jekyll-4.3.2/lib/jekyll/site.rb:191:in `each'
        from C:/Ruby31-x64/lib/ruby/gems/3.1.0/gems/jekyll-4.3.2/lib/jekyll/site.rb:191:in `generate'
        from C:/Ruby31-x64/lib/ruby/gems/3.1.0/gems/jekyll-4.3.2/lib/jekyll/site.rb:79:in `process'
        from C:/Ruby31-x64/lib/ruby/gems/3.1.0/gems/jekyll-4.3.2/lib/jekyll/command.rb:28:in `process_site'
        from C:/Ruby31-x64/lib/ruby/gems/3.1.0/gems/jekyll-4.3.2/lib/jekyll/commands/build.rb:65:in `build'
        from C:/Ruby31-x64/lib/ruby/gems/3.1.0/gems/jekyll-4.3.2/lib/jekyll/commands/build.rb:36:in `process'
        from C:/Ruby31-x64/lib/ruby/gems/3.1.0/gems/jekyll-4.3.2/lib/jekyll/command.rb:91:in `block in process_with_graceful_fail'
        from C:/Ruby31-x64/lib/ruby/gems/3.1.0/gems/jekyll-4.3.2/lib/jekyll/command.rb:91:in `each'
        from C:/Ruby31-x64/lib/ruby/gems/3.1.0/gems/jekyll-4.3.2/lib/jekyll/command.rb:91:in `process_with_graceful_fail'
        from C:/Ruby31-x64/lib/ruby/gems/3.1.0/gems/jekyll-4.3.2/lib/jekyll/commands/build.rb:18:in `block (2 levels) in init_with_program'
        from C:/Ruby31-x64/lib/ruby/gems/3.1.0/gems/mercenary-0.4.0/lib/mercenary/command.rb:221:in `block in execute'
        from C:/Ruby31-x64/lib/ruby/gems/3.1.0/gems/mercenary-0.4.0/lib/mercenary/command.rb:221:in `each'
        from C:/Ruby31-x64/lib/ruby/gems/3.1.0/gems/mercenary-0.4.0/lib/mercenary/command.rb:221:in `execute'
        from C:/Ruby31-x64/lib/ruby/gems/3.1.0/gems/mercenary-0.4.0/lib/mercenary/program.rb:44:in `go'
        from C:/Ruby31-x64/lib/ruby/gems/3.1.0/gems/mercenary-0.4.0/lib/mercenary.rb:21:in `program'
        from C:/Ruby31-x64/lib/ruby/gems/3.1.0/gems/jekyll-4.3.2/exe/jekyll:15:in `<top (required)>'
        from C:/Ruby31-x64/bin/jekyll:25:in `load'
        from C:/Ruby31-x64/bin/jekyll:25:in `<main>'
```

The _config file that was used contains the following. However removing the SQL queries didnt appear to prevent the crashes, so it doesn't appear to be a configuration issue:

```
sqlite:
  - data: brands
    file: _rims.sqlite
    query: SELECT * FROM brands
  - data: rims
    file: _rims.sqlite
    query: SELECT * FROM products
  - data: rims.vehicle_ids
    file: _rims.sqlite
    query: SELECT vehicle FROM vehicle_product_association WHERE product = :id
  - data: rims.vehicles
    file: _rims.sqlite
    query: SELECT * FROM vehicle WHERE id = :vehicle_ids

defaults:
  -
    scope:
      type: "pages"
      path: ""
    values:
      permalink: :path/:basename
      seo:
        description: "text"
  -
    scope:
      path: ""
    values:
      layout: "default"
      lang: "none"
  -
    scope:
      path: "nl"
    values:
      lang: "nl"
  -
    scope:
      path: "en"
    values:
      lang: "en"
  -
    scope:
      path: "de"
    values:
      lang: "de"
  -
    scope:
      path: "admin"
    values:
      loginRequired: true
  -
    scope:
      path: "*/email"
    values:
      layout: "email"
      sitemap: false
      hidden: true
  -
    scope:
      path: "assets/**/*.pdf"
    values:
      sitemap: false

sass:
  style: compressed

exclude:
  - composer.json
  - composer.lock
  - Gemfile
  - Gemfile.lock
  - uploads
  - vendor
  - minify.ps1

keep_files:
  - vendor
  - uploads

plugins:
  - jekyll-datapage-generator
  - jekyll_sqlite
```

**Comment by vaibhavb** [comment]
Any update? Is this plugin supported?

**Comment by captn3m0** [comment]
Apologies for the delay here - I'd been wanting to try the gem out in a new project to ensure everything was proper. I've published a new release `0.1.3` just now that gets the gem functional.

I also took the time to add some tests via the northwind database, with the configuration at https://github.com/captn3m0/jekyll-sqlite/blob/main/test/_config.yml.

If you face any further bugs or have feature suggestions, please do file a new issue.
